src/python/pants/backend/project_info/filter_targets.py

```diff
@@ -17,10 +18,16 @@
 from pants.util.filtering import and_filters, create_filters
 
 
+class TargetGranularity(Enum):
+    all_targets = "all"
+    file_targets = "file"
+    base_targets = "base"
```

Comment:
As mentioned on the ticket, feedback on the naming of both of these concepts is welcome.

"synthetic" targets are gone in v2, which might clear up the conceptual space to introduce the phrase "base". But it's possible that there is a better way to describe it.

Reply:
These names sound good to me. I've sometimes used "original" targets, but I like "base" more.

Reply:
Your help message alread has to translate base to BUILD in a parenthetical for the user - why not limit the nouns and stick with BUILD?

Reply:
"BUILD targets" vs. "file targets". That makes sense to me.

Reply:
Would it be capitalized in the enum? I'm not sure that it's clear what that means in the absence of the capitalization, and capitalizing one enum value would be odd.

Reply:
Yes, I'd capitalize it. And I'd trade odd for new word out of nowhere (for end users).

Reply:
Hm. Would we capitalize it in our APIs as well? i.e., would we rename `Address.is_base_target` to `Address.is_BUILD_target`? I'm skeptical that "build" is a distinctive enough word (although "base" isn't great either).

Reply:
Not sure and much less important since API users are a narrower set and better educated in Pants as a rule and API is not stabilised yet anyhow. I'd break off that question to other work.

Reply:
I think yes to lowercase build for code, but uppercase BUILD in docstring + website docs 

And we should probably stop using the name "generated subtargets" everywhere. Consistently use "BUILD targets" vs "file targets"

Reply:
I'm still not sure I agree, but it's not a huge deal. Will update. The resulting help string is:
```
                "Filter to rendering only BUILD targets (those declared in BUILD files), "
                 "only file-level targets, or all targets."
```
... because AFAIK it's still not clear what a "`BUILD` target" is without some explanation.

But on the `is_base_target` front we can invert the condition and replace it with the opposite method: `is_file_target`, so that concern is resolved at least.

Reply:
> ... because AFAIK it's still not clear what a "BUILD target" is without some explanation.

I agree given your wording, but with a slight tweak I think the parenthetical provides no more light:
"Filter to rendering only targets in BUILD files, only file-level targets, or all targets."